source/adapters/level_zero/queue.cpp

```diff
@@ -438,7 +438,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urQueueRelease(
       return Res;
 
     // Make sure all commands get executed.
-    Queue->synchronize();
+    UR_CALL(Queue->synchronize());
```

Comment:
I'm not sure if this isn't leaking the queue object. But other similar checks also don't do anything with it so dunno :)

Reply:
Yeah, I didn't see any cleanup in other parts of the code as well.

If we actually end up failing here (on this call or any other in this function) it will lead to abort on SYCL level because urQueueRelease is called inside a queue dtor and PI translates UR errors into exceptions (dtor is noexcept) so we don't need to worry about leaks. I don't know is this is desired but that's how it is right now. 

However, I guess it would nice to have some tests on UR level with error injection perhaps so we could check for leaks.